general: remove home-manager

tupakkatapa opened the pull request:
We'll be removing the home-manager module because it has an extensive number of dependencies that are unnecessary for a headless configuration.  Some refactoring and tweaks may also take place here.

tupakkatapa commented:
@jhvst Is it necessary to make this change in this repository, considering that it would only affect build server hosts? 

Home-manager is going to be removed from Ethereum-related hosts in https://github.com/ponkila/nixobolus/pull/9

jhvst commented:
Not necessary.

tupakkatapa commented:
> Not necessary.

NVM, there seem to be some changes that need to be made in this repository to adapt to the fact that home-manager is no longer available for the `core` user. I will make the necessary changes in this pull request.

tupakkatapa commented:
To rationalize the decision of removing home-manager, let's compare the differences for the `dinar-ephemeral-alpha` host:

With home-manager:

- Host size: 3.88 GiB
- Derivation tree: 32,672 lines

Without home-manager:

- Host size: 3.4 GiB
- Derivation tree: 29,561 lines

The size decreased by approximately 12.37%. Importantly, we do not lose any functionality by removing home-manager for this specific use case.

---
FYI: I've moved rust-motd to Hetzner hosts only. This is because there doesn't seem to be a clean way to retrieve only the usernames for `last_login` section without home-manager; the `config.users.users` contains a lot of unnecessary users.

I'll be adding the rust-motd for node hosts via Nixobolus, where I think it should be anyways.

jhvst commented:
Can you also check in the trees in this commit? Rather have them inspectable in git.

30k of packages sounds still like quite a lot.

tupakkatapa commented:
> To rationalize the decision of removing home-manager, let's compare the differences for the `dinar-ephemeral-alpha` host:

Updated with new method and added host of different format:

| Hostname                | Size (GiB) | Nix-store query (lines) |
|-------------------------|------------|------------------------|
| dinar-ephemeral-alpha   | 3.88 -> 3.40 | 4617 -> 4236  |
| ponkila-ephemeral-beta | 3.42 -> 2.68  | 4275 -> 3813 |

ref: https://github.com/ponkila/homestaking-infra/pull/45#issuecomment-1599346469

tupakkatapa commented on hosts/dinar-ephemeral-alpha/nix-store-query.txt:
```diff
@@ -4459,10 +4104,6 @@
 /nix/store/1fs906dngbqsscq0dcsij717bs3wkgj9-wasm-bindgen-shared-0.2.86.drv
 /nix/store/16yhqbv0grm1w01h6c8f9cscwcywja7b-wasm-streams-0.2.3.drv
 /nix/store/43sa68vwqdkp8jv8i60kqdavqsdjfykp-wasm-timer-0.2.5.drv
-/nix/store/4x7534qad2r4ik6rrfhg92qpy05y61qq-wayland-1.22.0.drv
-/nix/store/y4w36xa9z3zys2y9w0al3a5xb1k47w9n-wayland-1.22.0.tar.xz.drv
-/nix/store/1fg88nfcw8nwbmxmdhyqzay11fr90fr6-wayland-protocols-1.31.drv
-/nix/store/dkdaqs2s8a4k3fnl2iq9ss6lz599j5zk-wayland-protocols-1.31.tar.xz.drv
```
We got rid of wayland stuff
